manifests/istio-control/istio-discovery/templates/deployment.yaml

```diff
@@ -26,7 +29,9 @@ spec:
       maxUnavailable: {{ .Values.pilot.rollingMaxUnavailable }}
   selector:
     matchLabels:
-      {{- if ne .Values.version ""}}
+      {{- if .Values.global.revision }}
+      istio.io/rev: {{.Values.global.revision}}
+      {{- else if ne .Values.version ""}}
```

Comment:
apps/v1 Deployment label selectors are immutable. We'll need a new deployment for every revision bump. This is arguably a good thing for versioning multiple control planes. How will this in 1.5 where we really only support in-place upgrades?

Reply:
This should be a new deployment. I need to modify this PR to make it so its `istiod-{{.revision}}`, I forgot that somehow. in 1.5 we could remove this selector I guess?

Am I right in thinking that the 1.5 deployment (matching `app: pilot`) would match the 1.6 deployment (which has `app: pilot` as well as a revision label) and things will get weird

Reply:
`istiod-{{.revision}}` deployment naming to match the `istio.io/rev` label sounds good. 

I think we can remove the selector for 1.5. See my comment below about setting the istio.io/rev directly in the deployment below. 

I think you're right. 1.5 pods w/`app=pilot` would match the 1.6 deployments. If we changed the `app` label value that shouldn't be a problem though. s/pilot/istiod in 1.6 maybe?

Reply:
the only reason I kept it to pilot was to avoid issues where we had tooling selecting app=pilot (eg istioctl). But if we are already selecting on only that, we may not have much choice

Reply:
> I think you're right. 1.5 pods w/app=pilot would match the 1.6 deployments. 

I had this wrong. The k8s pilot service would match both. The replicasets versions shouldn't get mixed up. That is pretty much how our bookinfo demo is setup today.

Reply:
Interesting, seems like you can't actually run into the deployment label issue since replicaset adds a pod-hash label.

Let me fix up the names, service selectors, etc